adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```
@@ -1780,6 +1792,65 @@ private void checkInternetPermission() {
+            final PackageManager packageManager = mContext.getPackageManager();
+            final StringBuilder permissionMissing = new StringBuilder();
+            Logger.v(
+                    TAG + methodName,
+                    "Broker related permissions checking starts.");
+
+            if(PackageManager.PERMISSION_GRANTED != packageManager.checkPermission(
+                    "android.permission.GET_ACCOUNTS", mContext.getPackageName())) {
+                permissionMissing.append("GET_ACCOUNTS ");
+                Logger.w(
+                        TAG + methodName,
+                        "Broker related permissions are missing for GET_ACCOUNTS",
+                        "", ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING);
+            }
+
+            if(PackageManager.PERMISSION_GRANTED != packageManager.checkPermission(
+                    "android.permission.MANAGE_ACCOUNTS", mContext.getPackageName())) {
+                permissionMissing.append("MANAGE_ACCOUNTS ");
+                Logger.w(
+                        TAG + methodName,
+                        "Broker related permissions are missing for MANAGE_ACCOUNTS",
+                        "", ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING);
+            }
+
+            if(PackageManager.PERMISSION_GRANTED != packageManager.checkPermission(
+                    "android.permission.USE_CREDENTIALS", mContext.getPackageName())) {
+                permissionMissing.append("USE_CREDENTIALS");
+                Logger.w(
+                        TAG + methodName,
+                        "Broker related permissions are missing for USE_CREDENTIALS",
+                        "", ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING);
+            }
+
+            Logger.v(
+                    TAG + methodName,
+                    "Broker related permissions are verified");
+
+            if(permissionMissing.length() == 0) {
+                return true;
+            } else {
```

Do you mean the if-else is not needed? I think we should throw the exception if permission is missing, shouldn't we?